packages/toolpad-app/src/components/DialogForm.tsx

```diff
@@ -1,7 +1,39 @@
+import * as React from 'react';
 import { styled } from '@mui/material';
 
-export default styled('form')({
+const StyledDialogForm = styled('form')({
   overflowY: 'auto',
   display: 'flex',
   flexDirection: 'column',
 });
+
+export interface DialogFormProps extends React.FormHTMLAttributes<HTMLFormElement> {
+  children: React.ReactNode;
+  onKeyDown?: (event: React.KeyboardEvent) => void;
+}
+
+export default ({ children, onKeyDown = () => {}, ...rest }: DialogFormProps) => {
```

Comment:
[Upcoming eslint changes](https://github.com/mui/material-ui/pull/34642) will mandate function declarations. We can already adhere to this style

```suggestion
export default function ({ children, onKeyDown = () => {}, ...rest }: DialogFormProps) {
```